test/migrations/migrator.js

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import chai from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
+import {cleanDatabase, connectToMongo} from '../../src/utils/db_utils';
+import config from '../../config/config';
+import Migrator from '../../src/migrations/Migrator';
+import EmptyLogger from '../helpers/empty_logger';
+
+chai.use(sinonChai);
+chai.use(chaiAsPromised);
+const {expect} = chai;
+
+describe('Migrator', () => {
+  let db;
+  let client;
+  let migrator;
+  let logger;
+  let clock;
+  const migrationSleepTimeInSeconds = 3;
+  const now = 1500000;
+
+  before(async () => {
+    ({db, client} = await connectToMongo(config));
+    migrator = new Migrator(db, {
+      migrationSleepTimeInSeconds
+    });
+    logger = new EmptyLogger();
+  });
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers(now * 1000);
```

Comment:
This is no longer needed as you can now inject a sleep function that resolves immediately. 

`sinon.useFakeTimers` is evil 😈